lib/puppet/parser/functions/defined.rb

```diff
@@ -38,8 +38,30 @@
       defined(File[\'/some/file\'])
       defined(Class[\'foo\'])
 
+  When used with the future parser (4.x), the `defined` function does not answer if data
+  types (e.g. `Integer`) are defined, and the rules for asking for undef, empty strings, and
+  the main class are different:
```

Comment:
"does not answer" doesn't really say what the answer is and there's always an answer unless it raises an exception.